### ISSUE TITLE
Table Name Convention Using Two Words

### ISSUE THREAD

**Issue by dukesnuz** [susanBuck/dwa15-fall2017#201]
#### Scenario;
A table name consisting of two words. ie. email campaigns

#### Question
Would we use:
snake_case? email_campaigns
or
CamelCase? emailCampaigns

~~If we use snake_case, will this through an error when running migrations?
Reason I am asking is I used snake_case and getting an error when running ```migrate:fresh```
Error = ``` SQLSTATE[HY000]: General error: 1215 Cannot add foreign key constraint```~~
Update. My above error was caused by a typo. Still wondering about naming conventions

**Comment by ysrinu** [comment]
@dukesnuz  I have used snake_case for my table names.

While we should use naming standards as recommended by our professor, we should be able to work with any existing tablenames. Usually the DBAs will have their own naming standards that's most likely not follow our application framework guidelines.
I worked at a company where the DBAs implemented no-vowels in database and column names (and underscores). This was to keep the names terse and within the maximum length allowed.
In a database, the table and column names are always case-insensitive.

**Comment by susanBuck** [comment]
To find the answer to your question, I created a new Model called `EmailCampaign`. Then I used Eloquent's `getTable` method to see what it thinks the table name should be:

```php
$emailCampaign = new EmailCampaign();
dump($emailCampaign->getTable());
```

The results: `email_campaigns`

This can be overridden by explicitly setting the table name in the model:

```php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class EmailCampaign extends Model
{

    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'emailCampaigns';
}
```

If you didn't overwrite the default, and you were referencing this table via a foreign key, I would expect that that foreign key field should be called `email_campaigns_id`.  
This too, can be overwritten in the Model's relationship methods. Here are some examples ([borrowed from the docs](https://laravel.com/docs/5.5/eloquent-relationships#one-to-many)):

```php
return $this->hasMany('App\Comment', 'foreign_key');
```
or

```php
return $this->hasMany('App\Comment', 'foreign_key', 'local_key');
```

***

Side note: Whenever possible, I try to use single word table names to avoid confusion with pivot tables. For example, in this case I'd just call the table `campaigns`. It's not always possible though, so this is a good question.
